nlmod/plot/plot.py

```diff
@@ -477,7 +477,8 @@ def map_array(
 
     # select layer
     try:
-        nlay = da["layer"].shape[0]
+        # check if layer is a dimension
+        nlay = da["layer"].shape[0] if "layer" in da.dims else -1
```

Comment:
Is the `except KeyError` still necessary?

Reply:
Yes, this only checks if layer is in dims, if it is not in coords, the KeyError is raised.